src/lib/Imp.hs

```diff
@@ -507,6 +513,8 @@ toImpHof env (maybeDest, hof) = do
       copyAtom sDest =<< impSubst env s
       void $ translateBlock (env <> ref @> sDest) (Just aDest, body)
       PairVal <$> destToAtom aDest <*> destToAtom sDest
+    RunIO ~(Lam (Abs _ (_, body))) -> do
+      translateBlock env (maybeDest, body)
```

Comment:
You've ignored the dest! This might leave it uninitialized!

Reply:
Good catch!

Reply:
Actually I think it's fine. It gets passed to `translateBlock` just like in the `RunState` case.